listenbrainz/webserver/views/playlist_api.py

```diff
@@ -510,20 +510,23 @@ def get_playlist_xspf(playlist_mbid):
     """
 
     if not is_valid_uuid(playlist_mbid):
```

Comment:
There is one more case of exceptions that needs to handled, internal server errors. I think the easiest way to go about that would be to wrap the entire code in this function inside a try/except block and raise a PlaylistsAPIXMLError exception from there. Also, before you re-raise it log the exception using `current_app.logger.error("Failed to export playlist XSPF:", exc_info=True)`.

One common way this exception can happen is database connectivity issues, which is what you can do to test this case. Mock `db_playlist.get_by_mbid` and make it raise an exception.

Reply:
Sorry for the misleading comment, the new way is indeed the correct way and simplest way to do it.